core/server/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -565,7 +565,7 @@ public CreatePathResult createPath(LockedInodePath inodePath, CreatePathOptions<
           InodeDirectory.create(mDirectoryIdGenerator.getNewDirectoryId(),
               currentInodeDirectory.getId(), pathComponents[k], missingDirOptions);
       // Lock the newly created inode before subsequent operations, and add it to the lock group.
-      lockList.lockWrite(dir);
+      lockList.lockWriteAndCheckFullPath(dir, currentInodeDirectory, pathComponents[k]);
```

Comment:
Is it the case that, when locking an inode during an iteration, we ignore the returned `InvalidPathException`; while for all the other references, we escalate the exception? If yes, we should document it somewhere for future reference.